osu.Server.Spectator/Hubs/ScoreProcessedSubscriber.cs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Newtonsoft.Json;
+using osu.Framework.Logging;
+using osu.Game.Online.Spectator;
+using osu.Server.Spectator.Database;
+using StackExchange.Redis;
+using StatsdClient;
+using Timer = System.Timers.Timer;
+
+namespace osu.Server.Spectator.Hubs;
+
+public sealed class ScoreProcessedSubscriber : IScoreProcessedSubscriber, IDisposable
+{
+    /// <summary>
+    /// The maximum amount of time to wait for a <see cref="ScoreProcessed"/> message for a given score in milliseconds.
+    /// </summary>
+    private const int timeout_interval_ms = 30_000;
```

Comment:
We'll see how this plays out. I have a feeling going higher than this will be what we want eventually (minutes at least) to allow for potential server-side issues/outages.